src/main/webapp/components/navbar.vue

```diff
@@ -40,19 +43,67 @@ Vue.directive('click-outside', {
 module.exports = {
   data: function() {
     return {
-      showDropdown: false
+      showDropdown: false,
+      signedIn: false,
+      clientId: "506538592562-klrt6tseu4eg2pi7cvt9m84ifh5neccp.apps.googleusercontent.com",
+      googleAuth: null,
+      user: null
     }
   },
-  props: {
-    user: Object
-  },
   methods: {
     hideDropdown() {
       this.showDropdown = false;
     },
     toggleDropdown() {
       this.showDropdown = !this.showDropdown;
+    },
+    onGAPILoad() {
+      gapi.load('auth2', this.initGoogleAuth);
+    },
+    initGoogleAuth() {
+      gapi.auth2.init({
+        client_id: this.clientId
+      }).then(() => {
+        this.googleAuth = gapi.auth2.getAuthInstance();
+        if(this.googleAuth.isSignedIn.get()) this.fetchUser();
+      })
+    },
+    fetchUser() {
+      const user = this.googleAuth.currentUser.get();
+      const token = user.getAuthResponse().id_token;
+      fetch('/user-signin?idToken=' + token)
+        .then(response => response.json())
+        .then(userData => setUserInfo(userData))
+        .catch(err => {
+          console.log(err);
+        })
+    },
+    setUserInfo(userData) {
+      this.signedIn = true;
+      this.user = userData;
+      this.$emit('set-user', userData);
```

Comment:
Thanks for including this line, it should make it easier for the rest of the page to know which user is logged in. Alternatively, you could set a `watch` on `user` so that whenever user changes, that change is `$emit`-ed. This would also ensure that when the user signs out, that change is also emitted, which I don't currently see implemented.

watch example: https://stackoverflow.com/questions/34723680/vuejs-event-on-change-of-element-value

Reply:
Nice catch, I completely forgot about the logging out part! And I didn't know that there was a `watch` property, I'll add that instead